pkg/cloudprovider/providers/azure/azure_wrap.go

```diff
@@ -336,3 +336,28 @@ func convertResourceGroupNameToLower(resourceID string) (string, error) {
 	resourceGroup := matches[1]
 	return strings.Replace(resourceID, resourceGroup, strings.ToLower(resourceGroup), 1), nil
 }
+
+// isBackendPoolOnSameLB checks whether newBackendPoolID is on the same load balancer as existingBackendPools.
+// Since both public and internal LBs are supported, lbName and lbName-internal are treated as same.
+func isBackendPoolOnSameLB(newBackendPoolID string, existingBackendPools []string) (bool, string, error) {
```

Comment:
could you add comments about returned lbName, looks like the returned lbName depends on whether match or not, a little tricky.

Reply:
it is the lb name of existingBackendPools, if the codes go to the last line, then newLBName is actually same with lbName.

Reply:
Added the docs here.